libevmasm/Assembly.cpp

```diff
@@ -573,21 +573,21 @@ LinkerObject const& Assembly::assemble() const
 			"Cannot push and assign immutables in the same assembly subroutine."
 		);
 
-	size_t bytesRequiredForCode = bytesRequired(subTagSize);
+	unsigned bytesRequiredForCode = bytesRequired(static_cast<unsigned>(subTagSize));
 	m_tagPositionsInBytecode = vector<size_t>(m_usedTags, numeric_limits<size_t>::max());
 	map<size_t, pair<size_t, size_t>> tagRef;
 	multimap<h256, unsigned> dataRef;
 	multimap<size_t, size_t> subRef;
 	vector<unsigned> sizeRef; ///< Pointers to code locations where the size of the program is inserted
 	unsigned bytesPerTag = util::bytesRequired(bytesRequiredForCode);
-	uint8_t tagPush = (uint8_t)pushInstruction(bytesPerTag);
+	uint8_t tagPush = uint8_t(pushInstruction(bytesPerTag));
 
-	unsigned bytesRequiredIncludingData = bytesRequiredForCode + 1 + m_auxiliaryData.size();
+	unsigned bytesRequiredIncludingData = bytesRequiredForCode + 1 + static_cast<unsigned>(m_auxiliaryData.size());
```

Comment:
Well, yeah, that's what I'm afraid of here:
I'd argue this should rather be ``size_t`` instead.
The RHS here should be size_t, the additions in the loop below have ``size_t``s on their RHSs, ``util::bytesRequired`` can take a ``size_t`` and ``ret.bytecode.reserve`` gets a ``size_t``, so nothing lost and no cast.

On the other hand, strictly speaking we'd need bounds checks in the loop below or sth like that, if we cast to ``unsigned``...

How do we want to do this? If I review every change like that and we discuss each one of the, this will take forever... if we don't though, I wonder how much value is in ``-Wconversion``, if it just makes us use unsafe casts anyways...

Reply:
Although, maybe it won't be too bad in the end... let me have a look at the rest...

Reply:
Hm... I at least quickly looked through the rest, and I would have expected to see much more controversial cases... however, quite a few of them are cases of "it's reasonable assume that the value is small enough" without checking or "real" assurance... but maybe having the (oftentimes rather ugly) ``static_cast`` in those cases does have the benefit that one actually looks if the assumption is reasonable when reading the code... so maybe it's just fine...

Reply:
In short: the change is to match what was happening now. I did not dare to review the behaviour in many places, but arguably that's what we should do. I did clean it up in some places only.

Reply:
Hm... yeah... we did have some similar set of changes introducing ``static_cast``s some time ago, didn't we? With the signedness conversion? And I think there we basically said the same: make it work now and care in detail later... and probably never did care in the end ;-)? Maybe we should have an issue: ``look through each and every ``static_cast`` making sure their valid and trying to eliminate them some other way`` and do that some time...
But what's definitely true is that enabling the warnings will (hopefully) prevent us from continuing to introduce more such cases in the future, so maybe "just making it work", plus such an issue is ok.

Reply:
> But what's definitely true is that enabling the warnings will (hopefully) prevent us from continuing to introduce more such cases in the future, so maybe "just making it work", plus such an issue is ok.

Exactly, this is motivation: to stop this madness.

Hopefully someone, sometime will clean up the current mess too.